src/DynamoCore/Extensions/IExtensionStorageAccess.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Graph;
+
+namespace Dynamo.Extensions
+{
+    public interface IExtensionStorageAccess
+    {
+        /// <summary>
+        /// Action to be invoked when the workspace is opened. The passed extensionData dictionary is a copy of the <see cref="ExtensionData"/> data dictionary.
+        /// Modifying the extensionData from this action will not modify the stored <see cref="ExtensionData"/>.
+        /// To modify the stored <see cref="ExtensionData"/> data dictionary use the <see cref="OnWorkspaceSaving(Dictionary{string, string}, SaveContext)"/>.
+        /// </summary>
+        /// <param name="extensionData">A copy of the ExtensionData dictionary</param>
+        void OnWorkspaceOpen(Dictionary<string, string> extensionData);
+
+        /// <summary>
+        /// Action to be invoked when the workspace has begun its saving process.
+        /// The passed extensionData dictionary is a direct reference to the stored <see cref="ExtensionData"/> data dictionary, modifications to this dictionary
+        /// will be reflected in the stored <see cref="ExtensionData"/> data dictionary
+        /// </summary>
+        /// <param name="extensionData"></param>
+        /// <param name="saveContext"></param>
+        void OnWorkspaceSaving(Dictionary<string, string> extensionData, SaveContext saveContext);
```

Comment:
have you checked what happens during backup save? Does anything special need to be handled for that case?

Reply:
im not raising this on backups, same as with `OnSaved()`

https://github.com/DynamoDS/Dynamo/pull/11347/files#diff-14fa24535a2d0aa34f993496cc35eb02dbadcffb7576a54cbeef2e9821edba8cR558-R559